execution/executor-benchmark/src/transaction_generator.rs

```diff
@@ -680,12 +691,21 @@ impl TransactionGenerator {
 
         if transactions.is_empty() {
             let val = phase.fetch_add(1, Ordering::Relaxed);
+            let last_generated_at = last_non_empty_phase.load(Ordering::Relaxed);
+            if val > last_generated_at + 2 {
+                info!(
+                    "Block generation: no transactions generated in phase {}, and since {}, ending execution",
+                    val, last_generated_at
+                );
+                return true;
```

Comment:
Can you explain why this check is needed?

Reply:
For regular workloads,  all blocks have transactions. 

For workflows (i.e. staged workloads), we stop creating transactions when stage is done, and then empty block signals that new stage should start.  If we have two consecutive empty blocks, that means all stages are done. 

We generally don't know how many blocks need to be created, so in those cases we set large number of blocks (i.e 1000), and then use this to end benchmark when done.

Example - https://gist.githubusercontent.com/igor-aptos/149c414b95070f67989193012a21d7de/raw/62faa15aef3c67af23cfcdf637f4bcea42fda5c1/gistfile1.txt